src/agent/src/watcher.rs

```diff
@@ -87,6 +87,8 @@ async fn copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> Result<()> {
             fs::remove_file(&to).await?;
         }
         fs::symlink(fs::read_link(&from).await?, &to).await?;
+    } else if from.as_ref().is_dir() {
+        fs::create_dir_all(to).await?;
```

Comment:
Can we add/augment a unit test to exercise this?

Reply:
Yes, I'add.

Reply:
Great thanks. Any reason to not (also?) add to test_copy?  https://github.com/kata-containers/kata-containers/pull/3152/files#diff-3f11f988ef87c954cf00b1d8a6977cf2cd3cc6aac9514d3496b3bafd37258dfcR898

Reply:
@egernst There is no reason to do not to test `test_copy`, only because I missed it. Now added.